### PR TITLE
PC-4644 add changes to handle composite SLO

### PR DESCRIPTION
**Pull request by mkaras-nobl9**
Adds raw metrics on objective level

### CODE REVIEW

File: go.mod
Comment (mkaras-nobl9) [inline go.mod:8]: update with new version of nobl9-go with raw metric on objective level.

**Comment by mkaras-nobl9** [comment]
Acceptance tests result run on https://dev-performance.nobl9.dev with composite SLO enabled:

```
?   	github.com/nobl9/terraform-provider-nobl9	[no test files]
=== RUN   TestAcc_Nobl9SLO
=== PAUSE TestAcc_Nobl9SLO
=== CONT  TestAcc_Nobl9SLO
=== RUN   TestAcc_Nobl9SLO/test-prometheus
=== RUN   TestAcc_Nobl9SLO/test-prom-with-ap
=== RUN   TestAcc_Nobl9SLO/test-prom-with-countmetrics
=== RUN   TestAcc_Nobl9SLO/test-prom-with-multiple-objectives
=== RUN   TestAcc_Nobl9SLO/test-prom-full
=== RUN   TestAcc_Nobl9SLO/test-prom-with-time-slices
=== RUN   TestAcc_Nobl9SLO/test-prom-with-raw-metric-in-objective
=== RUN   TestAcc_Nobl9SLO/test-newrelic
=== RUN   TestAcc_Nobl9SLO/test-appdynamics
=== RUN   TestAcc_Nobl9SLO/test-splunk
=== RUN   TestAcc_Nobl9SLO/test-lightstep
=== RUN   TestAcc_Nobl9SLO/test-splunk-observability
=== RUN   TestAcc_Nobl9SLO/test-dynatrace
=== RUN   TestAcc_Nobl9SLO/test-thousandeyes
=== RUN   TestAcc_Nobl9SLO/test-graphite
=== RUN   TestAcc_Nobl9SLO/test-bigquery
=== RUN   TestAcc_Nobl9SLO/test-opentsdb
=== RUN   TestAcc_Nobl9SLO/test-multiple-ap
=== RUN   TestAcc_Nobl9SLO/test-composite-occurrences
=== RUN   TestAcc_Nobl9SLO/test-composite-time-slices
--- PASS: TestAcc_Nobl9SLO (915.26s)
    --- PASS: TestAcc_Nobl9SLO/test-prometheus (42.19s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-ap (41.64s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-countmetrics (51.21s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-multiple-objectives (53.38s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-full (39.98s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-time-slices (47.32s)
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-raw-metric-in-objective (40.99s)
    --- PASS: TestAcc_Nobl9SLO/test-newrelic (46.74s)
    --- PASS: TestAcc_Nobl9SLO/test-appdynamics (40.28s)
    --- PASS: TestAcc_Nobl9SLO/test-splunk (44.20s)
    --- PASS: TestAcc_Nobl9SLO/test-lightstep (41.03s)
    --- PASS: TestAcc_Nobl9SLO/test-splunk-observability (51.56s)
    --- PASS: TestAcc_Nobl9SLO/test-dynatrace (50.91s)
    --- PASS: TestAcc_Nobl9SLO/test-thousandeyes (42.83s)
    --- PASS: TestAcc_Nobl9SLO/test-graphite (50.46s)
    --- PASS: TestAcc_Nobl9SLO/test-bigquery (54.17s)
    --- PASS: TestAcc_Nobl9SLO/test-opentsdb (40.34s)
    --- PASS: TestAcc_Nobl9SLO/test-multiple-ap (41.77s)
    --- PASS: TestAcc_Nobl9SLO/test-composite-occurrences (51.83s)
    --- PASS: TestAcc_Nobl9SLO/test-composite-time-slices (42.44s)
PASS
ok  	github.com/nobl9/terraform-provider-nobl9/nobl9	916.545s
```

File: docs/index.md
Comment (lukasz-dobek) [inline docs/index.md:19]: This looks weird a bit weird imo, but I guess it's correct if it was generated by a plugin.

Comment (kskitek) [reply docs/index.md:19]: please revert it to be `nob9/nobl9`.

In case of self hosted provider, the source has to be an URL but in case of the official registry it has to be `{orgname}/{providername}` (exactly like with docker and dockerhub).
Our provider is published in the official registry and therefore should be `nobl9/nobl9` https://registry.terraform.io/providers/nobl9/nobl9/0.1.6

File: docs/resources/agent.md
Comment (kskitek) [inline docs/resources/agent.md:1]: Since it is not possible to create agents from sloctl / terraform on production, we should remove this file (see https://registry.terraform.io/providers/nobl9/nobl9/latest/docs that it is not there)

File: docs/index.md
Comment (kskitek) [inline docs/index.md:9]: can you please also update the version in the example to be `0.2.0`? I guess with CompositeSLOs we will release 0.2

File: Makefile
Comment (kskitek) [inline Makefile:6]: can you please also update the version in the example to be 0.2.0? I guess with CompositeSLOs we will release 0.2

**Review by kskitek — APPROVED** [review]
lgtm :rocket:

**Comment by ian-bartholomew** [comment]
Looks good, just a couple questions.  
- Looks like the tests are not passing in CI, is that because the environment its pointing to doesn't have the composite SLOs deployed to it?
- If the tests are breaking, does that mean that this is not currently backward compatible? Meaning if we were to make a release with it, would that break for customers?
Thanks!

**Comment by mkaras-nobl9** [comment]
- Yes. The composite SLOs is not on main yet, so I've used our dev environment for testing and posted results above.
- Yes. This will be communicated to users of the Tf provider with an email explaining why we need to do this.

**Comment by kskitek** [comment]
The change is not backwards compatible. It was discussed with our product. It was already communicated with some customers from what I have seen that there is a change coming and it will be communicated again when we have an ETA.

**Comment by RemiLis** [comment]
This should probably be fixed before merging (Terraform/terraform-provider-nobl9 — Build failed)
```
--- FAIL: TestAcc_Nobl9SLO (42.58s)                                                                                                                                                                                                                                                                                      
    --- FAIL: TestAcc_Nobl9SLO/test-prometheus (1.15s)                                                                                                                                                                                                                                                                   
    --- FAIL: TestAcc_Nobl9SLO/test-prom-with-ap (1.24s)                                                                                                                                                                                                                                                                 
    --- PASS: TestAcc_Nobl9SLO/test-prom-with-countmetrics (3.42s)                                                                                                                                                                                                                                                       
    --- FAIL: TestAcc_Nobl9SLO/test-prom-with-multiple-objectives (1.31s)                                                                                                                                                                                                                                                
    --- FAIL: TestAcc_Nobl9SLO/test-prom-full (1.23s)                                                                                                                                                                                                                                                                    
    --- FAIL: TestAcc_Nobl9SLO/test-prom-with-time-slices (1.23s)                                                                                                                                                                                                                                                        
    --- FAIL: TestAcc_Nobl9SLO/test-prom-with-raw-metric-in-objective (1.21s)                                                                                                                                                                                                                                            
    --- FAIL: TestAcc_Nobl9SLO/test-newrelic (1.27s)                                                                                                                                                                                                                                                                     
    --- FAIL: TestAcc_Nobl9SLO/test-appdynamics (1.17s)                                                                                                                                                                                                                                                                  
    --- FAIL: TestAcc_Nobl9SLO/test-splunk (1.15s)                                                                                                                                                                                                                                                                       
    --- FAIL: TestAcc_Nobl9SLO/test-lightstep (1.28s)                                                                                                                                                                                                                                                                    
    --- FAIL: TestAcc_Nobl9SLO/test-splunk-observability (1.15s)                                                                                                                                                                                                                                                         
    --- FAIL: TestAcc_Nobl9SLO/test-dynatrace (1.16s)                                                                                                                                                                                                                                                                    
    --- FAIL: TestAcc_Nobl9SLO/test-thousandeyes (16.36s)                                                                                                                                                                                                                                                                
    --- FAIL: TestAcc_Nobl9SLO/test-graphite (1.40s)                                                                                                                                                                                                                                                                     
    --- FAIL: TestAcc_Nobl9SLO/test-bigquery (1.41s)                                                                                                                                                                                                                                                                     
    --- FAIL: TestAcc_Nobl9SLO/test-opentsdb (1.34s)                                                                                                                                                                                                                                                                     
    --- FAIL: TestAcc_Nobl9SLO/test-multiple-ap (1.61s)                                                                                                                                                                                                                                                                  
    --- FAIL: TestAcc_Nobl9SLO/test-composite-occurrences (1.24s)                                                                                                                                                                                                                                                        
    --- FAIL: TestAcc_Nobl9SLO/test-composite-time-slices (1.24s)                                                                                                                                                                                                                                                        
FAIL                                                                                                                                                                                                                                                                                                                     
FAIL    github.com/nobl9/terraform-provider-nobl9/nobl9 61.264s                                                                                                                                                                                                                                                          
FAIL                                                                                                                                                                                                                                                                                                                     
make: *** [Makefile:45: testacc] Error 1
```

**Comment by ian-bartholomew** [comment]
@mkaras-nobl9 @kskitek yeah, i know that we communicated that there would be breaking changes when the backend is deployed, but since the backend that would make this work hasn't been deployed, we should hold off on merging this for now

**Comment by kskitek** [comment]
We are merging it now - the change is already on staging. New version will be registered in the official registry only on tag creation. We will tag new version when the change is on production.

**Pull request merged**
